tarantool/error.py

```diff
@@ -330,6 +330,33 @@ class PoolTolopogyError(DatabaseError):
     pass
 
 
+class CrudModuleError(DatabaseError):
+    """
+    Exception raised for errors that are related to the operation result of `crud`_ module.
+
+    .. _crud: https://github.com/tarantool/crud
+    """
+
+    def __init__(self, *args):
+        """
+        Sets fields with result and errors.
+        
+        :param args: The tuple from the crud module with result and errors.
+        :type args: :obj:`tuple`
+        """
+
+        # Sets tarantool.crud.CrudResult object.
+        self.res = args[0]
+
+        if isinstance(args[1], (tuple, list)):
+            # Case for crud_*_many().
+            # Sets list of tarantool.crud.CrudError objects.
+            self.errs = args[1]
+        else:
+            # Sets tarantool.crud.CrudError object.
+            self.err = args[1]
```

Comment:
Which fields do we have in CrudError? Can't we simply expose them here so it be similar to existing errors?

Reply:
It depends on the method we are calling. At the moment we are just exposing the fields from the Lua implementation (`args[0]` is `res` and `args[1]` is `err/errs` Lua vars). We can implement something like switch-case for each method, but is it necessary? Or did I misunderstand the question

Reply:
Let's mimic a `DatabaseError` in a simple way.

### What's in crud

#### For non-many
We have `res, err`, where `err` is a single error table from `errors` rock and `res` is `nil`.

#### For many
We have `res, err`, where `err` is an array of table errors and `res` is common result like in plain return.

### What's in python

`DatabaseError` is an `Exception` with `code`, `message` and `extra_info` (the same error, but with additional fields based on provided `box.error` info).

### How we can mix it

First case is rather simple. We can raise some error with the same code and message we've got from Tarantool. But since we've got additional info, it would be nice to provide it.

Second case is much complicated. My proposal is as follows:

- Have two different classes for non-many and many errors (`CrudModuleError` and `CrudModuleManyError`).
- Both should inherit `DatabaseError`.

For `CrudModuleError`, let's add the `extra_info_errors` attribute with the original error (`DatabaseError` has `extra_info` field with similar meaning, but different structure).

For `CrudModuleManyError`, let's raise error based on `DatabaseError(0, "Got multiple errors, see errors_list")`, store original errors list in `errors_list` attribute and success tuples in `success_tuples` attribute (I'm not insist on naming here -- it's just an example, but I think the current `res`, `errs` one is a bit confusing).

You also may define a new class for `errors` Lua rock errors since they always have the same structure, but this is optional -- dict is fine too.

Reply:
done

Reply:
Done, I also updated the documentation in `quick-start.rst`